service.autosubs/default.py

```diff
@@ -0,0 +1,161 @@
+#!/usr/bin/python
+
+import os
+import sys
+import xbmc
+import xbmcaddon
+
+__addon__ = xbmcaddon.Addon()
+__author__ = __addon__.getAddonInfo('author')
+__scriptid__ = __addon__.getAddonInfo('id')
+__scriptname__ = __addon__.getAddonInfo('name')
+__cwd__ = __addon__.getAddonInfo('path')
+__version__ = __addon__.getAddonInfo('version')
+__language__ = __addon__.getLocalizedString
+debug = __addon__.getSetting("debug")
+__cwd__ = xbmc.translatePath(__addon__.getAddonInfo('path'))
+__profile__ = xbmc.translatePath(__addon__.getAddonInfo('profile'))
+__resource__ = xbmc.translatePath(os.path.join(__cwd__, 'resources'))
+
+__settings__ = xbmcaddon.Addon("service.autosubs")
+
+ignore_words = (__settings__.getSetting('ignore_words').split(','))
+ExcludeTime = int((__settings__.getSetting('ExcludeTime'))) * 60
+
+currentPlaying = ""
+
+sys.path.append(__resource__)
+
+
+def Debug(msg, force=False):
+    if (debug == "true" or force):
+      print ("#####[AutoSubs]##### " + msg)
+
+Debug("Loading '%s' version '%s'" % (__scriptname__, __version__))
+
+
+# helper function to get string type from settings
+def getSetting(setting):
+    return __addon__.getSetting(setting).strip()
+
+
+# helper function to get bool type from settings
+def getSettingAsBool(setting):
```

Comment:
please use the new methods in our python api: https://codedocs.xyz/xbmc/xbmc/group__python__xbmcaddon.html

we have `xbmcaddon.Addon([id]).getSettingInt(id)`, `xbmcaddon.Addon([id]).getSettingBool(id)` and so on

Reply:
ok